airflow/models/dagrun.py

```diff
@@ -980,11 +980,11 @@ def create_ti(task: "Operator", indexes: Tuple[int, ...]) -> Generator:
     def _create_tasks(
         self,
         dag: "DAG",
-        task_creator: Callable,
-        task_filter: Callable,
+        task_creator: Callable[[Operator, Tuple[int, ...]], Union[Iterator[Dict[str, Any]], Iterator[TI]]],
+        task_filter: Callable[[Operator], bool],
         *,
         session: Session,
-    ) -> Iterable["Operator"]:
+    ) -> Union[Iterator[Dict[str, Any]], Iterator[TI]]:
```

Comment:
And this can be parametrized...

```python
CreatedTasksType = TypeVar("CreatedTasksType")

def _create_tasks(
    self,
    dag: "DAG",
    task_creator: Callable,
    task_filter: Callable,
    task_creator: Callable[[Operator, Tuple[int, ...]], CreatedTasksType,
    task_filter: Callable[[Operator], bool],
    *,
    session: Session,
) -> CreatedTasksType:
```

to avoid the `type: ignore` comments (I believe)

Reply:
Alright makes sense. But the arguments should occur only once right?

Reply:
Oh yes, I copy-pasted them from the diff incorrectly 😓